File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited=new Array(n).fill(false);
    
    function dfs(node){
        visited[node]=true;
        for(let i=0; i<n; i++){
            if(computers[node][i]===1&&!visited[i]){
                dfs(i);
            }
        }
    }
    
    
    for(let i=0; i<n; i++){
        if(!visited[i]){
            dfs(i);
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    static boolean[] visited;
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited=new boolean[n];
        
        for(int i=0; i<n; i++){
           if(!visited[i]){
               dfs(i,computers,n);
               answer++;
           }
        }
        return answer;
    }
    
    public static void dfs(int curi,int[][] computers,int n){
    
        visited[curi]=true;
        
        for(int i=0; i<n; i++){
            if(!visited[i]&&computers[curi][i]==1){
                dfs(i,computers,n);
            }
        }
    }
}